core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -44,11 +44,12 @@ func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceCo
 	}
 
 	return &OutgoingConnectorHandler{
-		gc:          gc,
-		method:      method,
-		responses:   newResponses(),
-		rateLimiter: rateLimiter,
-		lggr:        lgger,
+		gc:              gc,
+		gatewaySelector: NewRoundRobinSelector(gc.GatewayIDs()),
```

Comment:
@MStreet3 Thinking about this a bit more -- Is this safe? It basically queries GatewayIDs once, at instantiation time; I wonder if there could be some weird races where we haven't connected to everything and so our list is shorter than it otherwise could be. Maybe it's worth passing in the list of IDs every time we make a choice?

Reply:
Actually 🤦 nvm, you can see that the size of the list doesn't change here: https://github.com/smartcontractkit/chainlink/blob/331a2e829e836f700a48aebc23820f14032471d9/core/services/gateway/connector/connector.go#L129

Reply:
The reason I get the list at creation time is so that the index of the round robin increments correctly.  But in retrospect that doesn't have to be the case.  Since (as you point out) the list is fixed anyway I'm leaving it as is.